types: correct atttypmod for DECIMAL with precision but no width

otan opened the pull request:
Resolves #71926

Release note (bug fix): Previously, atttypmod in
pg_catalog.pg_attributes for DECIMAL types with precision but no
width was incorrectly -1. This is now populated correctly.

rafiss reviewed (approved):
so sneaky! nit on the commit message

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72016#-:-Mn-R8GMBjcxprXuV7gD:bi4g2tg)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @rafiss)

___
*[-- commits, line 4 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/72016#-Mn-QKir70Cm5IzMgvm1):*
nit: let's make this a bug fix and also s/atttypmod/pg_attribute.atttypmod


<!-- Sent from Reviewable.io -->

otan committed:
types: correct atttypmod for DECIMAL with precision but no width

Release note (bug fix): Previously, atttypmod in
pg_catalog.pg_attributes for DECIMAL types with precision but no
width was incorrectly -1. This is now populated correctly.
